clusterman/draining/queue.py

```diff
@@ -91,9 +92,16 @@ def __init__(self, cluster_name: str) -> None:
             "global_redraining_delay_seconds",
             default=DEFAULT_GLOBAL_REDRAINING_DELAY_SECONDS,
         )
+        self.resource_group_cache_seconds = staticconf.read_int(
+            f"clusters.{cluster_name}.resource_group_cache_seconds",
+            default=DEFAULT_RESOURCE_GROUPS_CACHE_SECONDS,
+        )
         self.drain_queue_url = staticconf.read_string(f"clusters.{cluster_name}.drain_queue_url")
         self.termination_queue_url = staticconf.read_string(f"clusters.{cluster_name}.termination_queue_url")
         self.draining_host_ttl_cache: Dict[str, arrow.Arrow] = {}
+        self.spot_fleet_resource_groups_cache: Set[str] = set()
+        self.autoscaling_resource_groups_cache: Set[str] = set()
+        self.resource_group_cache_expiration_time: arrow.Arrow = arrow.now()
```

Comment:
This should be set at time 0, so you are sure to do a first pass.
That said, since cachetools is already installed for the repo, I'd honestly just move each set of IDs to it's own time-cached property with separate TTLs (as we can make it a bit longer for ASGs).